### DISCUSSION TITLE
how to change primay color in runtime ?

### DISCUSSION THREAD

**Discussion by vaynevayne** [heroui-inc/heroui#1229] [Q&A]
same as title

**Comment by lveillard** [comment]
+1 to any solution to be able to apply some runtime styling to specific components)

**Comment by jrgarciadev** [comment]
Hey guys to change the theme on runtime you have to change the theme name class, please check the [Dark mode](https://nextui.org/docs/customization/dark-mode) documentation.

**Reply by vaynevayne** [reply]
no, I am not referring to dark mode, but changing the theme color, for example, in light mode, changing the theme color to red, green, etc
Depending on the tailwindcss library, I find it very difficult and I have no idea, so I'm asking for your advice. @jrgarciadev

**Reply by lveillard** [reply]
Same. Dark theme is not modified in runtime, it is compiled in build time.

We are looking for runtime style for instance imagine a client side color picker, or a color code stored in database that would apply to all components as primary color.

In my case i would also love to have other runtime styles like for instance being able to modify the margin. Something like what this lib does: https://twind.dev/ which actually works at runtime even if it uses tailwind utility classes.

**Comment by vaynevayne** [comment]
It doesn't seem to be supported right now, but modifying it here will do it, does it need pr?

<img width="583" alt="image" src="https://github.com/nextui-org/nextui/assets/124666577/b5ee6f6d-1519-4bf8-97e1-cb5569533fdf">

**Comment by renhiyama** [comment]
please open it back again
